toolchain/lowering/lowering_context.h

```diff
@@ -84,8 +99,12 @@ class LoweringContext {
   // a map; a map may end up lower memory consumption, but a vector offers cache
   // efficiency and better performance. As a consequence, the right choice is
   // unclear.
-  llvm::SmallVector<llvm::PointerUnion<llvm::Type*, llvm::Value*>>
+  llvm::SmallVector<
+      llvm::PointerUnion<llvm::FunctionType*, llvm::Type*, llvm::Value*>>
       lowered_nodes_;
+
+  // Maps callables to lowered functions.
+  std::vector<llvm::Function*> lowered_callables_;
```

Comment:
Maybe use `SmallVector` just for consistency here unless there is a reason? (Again, minor point and no perf concerns, just consistency.)

Reply:
Fixing